programs/drift/src/math/margin.rs

```diff
@@ -296,6 +347,51 @@ pub fn calculate_margin_requirement_and_total_collateral_and_liability_info(
             let token_value =
                 get_strict_token_value(token_amount, spot_market.decimals, &strict_oracle_price)?;
 
+            let mut updated_token_amount = token_amount;
```

Comment:
think it'd be nice to hide some of this logic in margin calculation struct so this code is easier to read